openapi/components/schemas/ExternalIdentifier.yaml

```
@@ -0,0 +1,47 @@
+type: object
+required:
+  - externalId
+properties:
+  resource:
+    type: string
+    description: Type of the resource.
+    enum:
+      - customers
+      - invoices
+      - invoice-items
+      - transactions
+      - journal-accounts
+      - journal-entries
+    readOnly: true
+  resourceId:
+    type: string
+    description: ID of the resource entity.
+    readOnly: true
+  service:
+    type: string
+    description: Name of the external service.
+    enum:
+      - quickbooks-online
+    readOnly: true
+  externalId:
+    type: string
+    description: ID of the referenced entity within an external service.
+  createdTime:
+    $ref: ./CreatedTime.yaml
+  updatedTime:
+    $ref: ./UpdatedTime.yaml
+  _links:
+    type: array
```

nitpick: this was displayed as a blank description in the rendered docs
```suggestion
  _links:
    description: Related links.
    type: array
```